### PR TITLE
chore: replace `mocha-circleci-reporter` with `mocha-multi-reporters`

### PR DESCRIPTION
**Pull request by luvkapur**
This PR replaces the `mocha-circleci-reporter` (https://github.com/sandcastle/mocha-circleci-reporter) which is no longer maintained and has a security vulnerability with `mocha-multi-reporters`, that uses `mocha-junit-reporter` to output to xml and `spec` to output to console.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -895,12 +895,13 @@ jobs:
...
           name: 'Run e2e tests'
-          command: 'cd bit && circleci tests glob "e2e/**/*.e2e*.ts" | circleci tests split --split-by=filesize | xargs -n 1 npm run mocha-circleci'
+          command: cd bit && npm run e2e-test-circle
+          # command: 'cd bit && circleci tests glob "e2e/**/*.e2e*.ts" | circleci tests split --split-by=filesize | xargs -n 1 npm run mocha-circleci'
           # command which support only - for debug purpose
           # command: cd bit && npm run e2e-test-circle --debug
           environment:
             MOCHA_FILE: junit/e2e-test-results.xml
-            NODE_OPTIONS: --no-warnings --max-old-space-size=5000
+            # NODE_OPTIONS: --no-warnings --max-old-space-size=5000
```
Comment (GiladShoham) [inline .circleci/config.yml:904]: @luvkapur why did you remove also the max old space?

Comment (luvkapur) [reply .circleci/config.yml:904]: because I saw we pass it as part of the script for this command
`mocha-circleci": "cross-env NODE_OPTIONS='--no-warnings --max-old-space-size=5000'`
I see it is missing here for the e2e-test-circle cmd which I overlooked 
`e2e-test-circle": "cross-env NODE_OPTIONS=--no-warnings`
i will add it there and then we dont need it as part of config.yml since it is redundant

**Review by davidfirst — APPROVED** [review]
